src/core/sources/WebRTC.ts

```diff
@@ -103,38 +135,37 @@ export const RoomParticipant = {
         )
         previousTracks = tracks
 
-        updateParticipants()
-
         newTracks.forEach((x) => {
-          const { mediaStreamTrack, id, participantId, type } = room.getTrack(
-            x.id,
-          )
-          const source = {
-            id,
-            isActive: true,
-            value: mediaStreamTrack,
-            props: {
-              trackId: id,
-              participantId,
-              isMuted: x.isMuted,
-              type,
-            },
-          } as Compositor.Source.NewSource
-
-          // Add each new track as a source
-          if (mediaStreamTrack) {
-            if (mediaStreamTrack.kind === 'video') {
-              addSource(source)
-            } else {
-              addSource(source)
-            }
+          const { id, participantId, type, isExternal } = room.getTrack(x.id)
+          if (type === 'camera') {
+            const { displayName, meta } = room.getParticipant(participantId)
+
+            const srcObject = new MediaStream([])
+            participantStreams[id] = srcObject
```

Comment:
Why is `srcObject` simply a `new MediaStream`? why not use the `mediaStreamTrack` from `room.getTrack`

Reply:
The final result used by weebrtc transformer is an srcObject. We update the srcobject with  mediastreamtrack when calling updateParticipant.